msprime/intervals.py

```diff
@@ -347,3 +310,65 @@ def get_length(self):
 
     def asdict(self):
         return self.map.asdict()
+
+
+def read_hapmap(filename):
+    # Black barfs with an INTERNAL_ERROR trying to reformat this docstring,
+    # so we explicitly disable reformatting here.
+    # fmt: off
+    """
+    Parses the specified file in HapMap format. These files must be
+    white-space-delimited, and contain a single header line (which is
+    ignored), and then each subsequent line contains the starting position
+    and recombination rate for the segment from that position (inclusive)
+    to the starting position on the next line (exclusive). Starting
+    positions of each segment are given in units of bases, and
+    recombination rates in centimorgans/Megabase. The first column in this
+    file is ignored, as are additional columns after the third (Position is
+    assumed to be the second column, and Rate is assumed to be the third).
+    If the first starting position is not equal to zero, then a
+    zero-recombination region is inserted at the start of the chromosome.
+
+    A sample of this format is as follows::
+
+        Chromosome	Position(bp)	Rate(cM/Mb)	Map(cM)
+        chr1	55550	        2.981822	0.000000
+        chr1	82571	        2.082414	0.080572
+        chr1	88169	        2.081358	0.092229
+        chr1	254996	        3.354927	0.439456
+        chr1	564598	        2.887498	1.478148
+        ...
+        chr1	182973428	2.512769	122.832331
+        chr1	183630013	0.000000	124.482178
+
+    :param str filename: The name of the file to be parsed. This may be
+        in plain text or gzipped plain text.
+    """
+    # fmt: on
+    positions = []
+    rates = []
+    if filename.endswith(".gz"):
+        f = gzip.open(filename)
+    else:
+        f = open(filename)
+    try:
+        # Skip the header line
+        f.readline()
+        for j, line in enumerate(f):
+            pos, rate = map(float, line.split()[1:3])
+            if j == 0:
+                map_start = pos
+                if pos != 0:
+                    positions.append(0)
+                    rates.append(0)
+            positions.append(pos)
+            # Rate is expressed in centimorgans per megabase, which
+            # we convert to per-base rates
+            rates.append(rate * 1e-8)
+        if rate != 0:
+            raise ValueError(
+                "The last rate provided in the recombination map must be zero"
+            )
+    finally:
+        f.close()
+    return RecombinationMap(positions, rates, map_start=map_start)
```

Comment:
We should return a RateMap here, rather than a RecombationMap (which is going for full deprecation)

Reply:
There's the small issue of the `map_start` parameter. How do you want to handle that? Add that to the rate map?

Reply:
Ah, yes. Yes, I guess we should add it to the RateMap all right.